runners/jstorm/src/main/java/com/alibaba/jstorm/beam/translation/translator/ViewTranslator.java

```diff
@@ -0,0 +1,374 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alibaba.jstorm.beam.translation.translator;
+
+import com.alibaba.jstorm.beam.translation.TranslationContext;
+import com.alibaba.jstorm.beam.translation.runtime.ViewExecutor;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.CoderRegistry;
+import org.apache.beam.sdk.coders.KvCoder;
+import org.apache.beam.sdk.coders.ListCoder;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.View;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionView;
+import org.apache.beam.sdk.values.PCollectionViews;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A {@link TransformTranslator} for executing {@link View Views} in JStorm runner.
+ */
+public class ViewTranslator extends TransformTranslator.Default<ViewTranslator.CreateJStormPCollectionView<?, ?>> {
+    @Override
+    public void translateNode(CreateJStormPCollectionView<?, ?> transform, TranslationContext context) {
+        TranslationContext.UserGraphContext userGraphContext = context.getUserGraphContext();
+        String description = describeTransform(transform, userGraphContext.getInputs(), userGraphContext.getOutputs());
+        ViewExecutor viewExecutor = new ViewExecutor(description, userGraphContext.getOutputTag());
+        context.addTransformExecutor(viewExecutor);
+    }
+
+    /**
+     * Specialized implementation for
+     * {@link org.apache.beam.sdk.transforms.View.AsMap View.AsMap}
+     * for the Flink runner in streaming mode.
+     */
+    public static class ViewAsMap<K, V>
+            extends PTransform<PCollection<KV<K, V>>, PCollectionView<Map<K, V>>> {
+
+        @SuppressWarnings("unused") // used via reflection in JstormRunner#apply()
+        public ViewAsMap(View.AsMap<K, V> transform) {
+        }
+
+        @Override
+        public PCollectionView<Map<K, V>> expand(PCollection<KV<K, V>> input) {
+            PCollectionView<Map<K, V>> view =
+                    PCollectionViews.mapView(
+                            input,
+                            input.getWindowingStrategy(),
+                            input.getCoder());
+
+            @SuppressWarnings({"rawtypes", "unchecked"})
+            KvCoder<K, V> inputCoder = (KvCoder) input.getCoder();
+            try {
+                inputCoder.getKeyCoder().verifyDeterministic();
+            } catch (Coder.NonDeterministicException e) {
+                // TODO: log warning as other runners.
+            }
+
+            return input
+                    .apply(Combine.globally(new Concatenate<KV<K, V>>()).withoutDefaults())
```

Comment:
This expansion is the same as what the SDK already provides. I think you can see new changes to the other runners that simplify this.